### ISSUE TITLE
Using nomad namespace causes inconsistent result after apply error

### ISSUE THREAD

**Issue by JanBerktold** [terraform-providers/terraform-provider-nomad#69]
When setting a nomad namespace, the provider _always_ produces an `Provider produced inconsistent result after apply` error. The example can be fixed by setting the `namespace` input to an empty string.
 
## Terraform Version
```
Terraform v0.12.4
+ provider.nomad v1.4.0
```

### Affected Resource(s)
- nomad_job

### Terraform Configuration Files
```hcl
resource "nomad_job" "memcached-job" {
  jobspec = templatefile(
    "${path.module}/job-spec.tpl",
    {
      name            = "testing-job"
      namespace       = "caching-platform"
    }
  )
}
```

The template:
```hcl
job "${name}" {
  %{ if namespace != "" }
  namespace = "${namespace}"
  %{ endif } 
  datacenters = ["alpha"]
  type = "service"

  group "cache" {
    count = 1
    task "redis" {
      driver = "docker"
      config {
        image = "redis:3.2"
        port_map {
          db = 6379
        }
      }
    }
  }
}
```

### Debug Output
https://gist.github.com/JanBerktold/cc66903f5a30d118be34da42ca727851

### Panic Output
Not applicable.

### Expected Behavior
Expected Nomad job to run in namespace.

### Actual Behavior
```
Error: Provider produced inconsistent result after apply

When applying changes to nomad_job.memcached-job, provider "nomad" produced an
unexpected new value for was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.
```

### Steps to Reproduce
1. `terraform apply`

**Comment by cgbaker** [comment]
Thanks for the report, @JanBerktold , working on this now.

**Comment by cgbaker** [comment]
The root cause seems to be that the calls to the Nomad API (specifically read and delete) do not reference the namespace. One limited workaround is to set the NOMAD_NAMESPACE environment variable before calling `terraform plan/apply/destroy/...`. This workaround assumes that the operator workflow supports that; it also assumes that the Terraform spec only references a single namespace.

I will release a 1.4.1 provider when this is fixed. I'm updating the [Nomad Job datasource](https://www.terraform.io/docs/providers/nomad/d/job.html) as well.

**Comment by cgbaker** [comment]
@JanBerktold , just released 1.4.1 with support for this. let me know how it works.

**Comment by robloxrob** [comment]
Thank you for the quick turn around!
